src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use the destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ...... } = action;
    
    switch (type) {
```